lib/srv/server/installer/defaultinstallers.go

```diff
@@ -26,19 +26,56 @@ import (
 	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
 )
 
-// DefaultInstaller represents the default installer script provided by teleport.
-var DefaultInstaller = oneoffScriptToDefaultInstaller()
+const (
+	scriptShebangAndSetOptions = `#!/usr/bin/env sh
+set -euo pipefail`
+	execGenericInstallScript = `
+INSTALL_SCRIPT_URL="https://{{.PublicProxyAddr}}/scripts/install.sh"
 
-func oneoffScriptToDefaultInstaller() *types.InstallerV1 {
-	argsList := []string{
+echo "Offloading the installation part to the generic Teleport install script hosted at: $INSTALL_SCRIPT_URL"
+
+TEMP_INSTALLER_SCRIPT="$(mktemp)"
+curl -sSf "$INSTALL_SCRIPT_URL" -o "$TEMP_INSTALLER_SCRIPT"
+
+chmod +x "$TEMP_INSTALLER_SCRIPT"
+
+sudo "$TEMP_INSTALLER_SCRIPT" || (echo "The install script ($TEMP_INSTALLER_SCRIPT) returned a non-zero exit code" && exit 1)
+rm "$TEMP_INSTALLER_SCRIPT"`
+)
+
+// LegacyDefaultInstaller represents the default installer script provided by teleport.
+var (
+	// LegacyDefaultInstaller uses oneoff.sh to download the Teleport tarball and run `teleport install`.
+	// The Teleport install command handles both Teleport installation and agent configuration.
+	LegacyDefaultInstaller = oneoffScriptToDefaultInstaller()
+
+	// NewDefaultInstaller installs Teleport by calling the standard "/scripts/install.sh" route on the proxy.
+	// After successfully installing Teleport, it will invoke the same `teleport install`
+	// command as the LegacyDefaultInstaller which will only take care of configuring Teleport.
+	NewDefaultInstaller = types.MustNewInstallerV1(
```

Comment:
Should we have some test for the rendered default installer given some template args?